### DISCUSSION TITLE
How does autoseg3d handle empty ground truth?

### DISCUSSION THREAD

**Discussion by paulwrightwork** [Project-MONAI/MONAI#7389] [Q&A] [answered]
Hi autoseg3d geniuses! I have a question as a relative newbie.

My overarching question is whether including images that should be empty in my training set will improve the trained model in autoseg3d. In my case I'm segmenting lesions so my dataset includes healthy images with no lesion. My fuzzy understanding is that including these should help the model learn what a normal background image looks like so it should find fewer false positives etc. Is this correct?

On the technical level, I understand how Dice works, and it's not that great for indicating whether the prediction is good when the ground truth is empty. I'm told autoseg3d uses cross entropy loss, which I haven't learned yet, but am told that it takes into account correctly labelled background voxels. I'd just like to hear from the horse's mouth how the losses are implemented in autoseg3d and whether my fuzzy understanding is fuzzily correct.

Many thanks.

**Comment by KumoLiu** [comment, marked as answer]
Hi @pwrightkcl, the foreground and background in segmentation represent positive and negative samples, respectively. Of course, you are correct that entering negative samples for the network is beneficial, but more is not always better, so when we have a relatively small proportion of foreground, we will also try to sample more positive samples or use weighted losses to improve the training.
https://github.com/Project-MONAI/research-contributions/blob/2449b45dabd859afa21a4dbba8f8f583f40c7920/auto3dseg/algorithm_templates/dints/configs/hyper_parameters.yaml#L56
https://github.com/Project-MONAI/research-contributions/blob/2449b45dabd859afa21a4dbba8f8f583f40c7920/auto3dseg/algorithm_templates/swinunetr/configs/hyper_parameters.yaml#L54

**Reply by paulwrightwork** [reply]
Thanks @KumoLiu that is educational for me. I'm using segresnet for my model and those examples are for dints and swinunetr. Does segresnet do something similar? (I plan to do a model selection run, but segresnet is my starting point).

**Reply by KumoLiu** [reply]
Yes, you can find the hyperparameter here.
https://github.com/Project-MONAI/research-contributions/blob/2449b45dabd859afa21a4dbba8f8f583f40c7920/auto3dseg/algorithm_templates/segresnet/configs/hyper_parameters.yaml#L57

**Comment by landynfrancis** [comment]
Does the DINTS model have different behavior? I am noticing I get an error 
`RuntimeError: grid_sampler(): expected input to have non-empty spatial dimensions, but input has sizes [1, 1, 0, 0, 0] with dimension 2 being empty`
When my training gets to an image/label pair where the label is empty.
